src/core_tests/determinism.rs

```diff
@@ -0,0 +1,94 @@
+use crate::{
+    pollers::MockServerGatewayApis,
+    prototype_rust_sdk::{TestRustWorker, WfContext, WorkflowResult},
```

Comment:
That's what it is - it will be separated into its own crate but it's the API users writing Rust workflows would use. It happens to be the easiest way to write most tests too. Hence Rust SDK rather than Core